candle-examples/examples/llava/main.rs

```diff
@@ -0,0 +1,252 @@
+    debug: bool, // now useless
+    #[arg(long, action)]
+    cpu: bool,
+    #[arg(long, action)]
+    no_kv_cache: bool,
+    #[arg(long, default_value = "Is this a cat?")]
+    prompt: String,
+    /// The seed to use when generating random samples. Copy from candle llama. Not exist in python llava.
+    #[arg(long, default_value_t = 299792458)]
+    seed: u64,
+}
+
+//from https://github.com/huggingface/candle/blob/main/candle-examples/examples/clip/main.rs
+fn load_image<T: AsRef<std::path::Path>>(
+    path: T,
+    processor: &CLIPImageProcessor,
+    llava_config: &LLaVAConfig,
+    dtype: DType,
+) -> anyhow::Result<((u32, u32), Tensor)> {
+    let img = image::io::Reader::open(path)?.decode()?;
+    let img_tensor = process_image(&img, processor, llava_config)?;
+    Ok(((img.width(), img.height()), img_tensor.to_dtype(dtype)?))
+}
+
+fn get_model_name(path: &str) -> String {
+    path.split('/').last().unwrap().to_string()
+}
+
+fn main() -> Result<()> {
+    let mut args = Args::parse();
+    let device = candle_examples::device(args.cpu)?;
+    println!("Start loading model");
+    let api = Api::new()?;
+    let api = api.model(args.model_path.clone());
+    let model_name = get_model_name(&args.model_path);
+    let (llava_config, tokenizer, clip_vision_config, image_processor) = if model_name
+        .contains("hf")
+    {
+        let config_filename = api.get("config.json")?;
+        let hf_llava_config: HFLLaVAConfig =
+            serde_json::from_slice(&std::fs::read(config_filename)?)?;
+        let generation_config_filename = api.get("generation_config.json")?;
+        let generation_config: HFGenerationConfig =
+            serde_json::from_slice(&std::fs::read(generation_config_filename)?)?;
+        let preprocessor_config_filename = api.get("preprocessor_config.json")?;
+        let preprocessor_config: HFPreProcessorConfig =
+            serde_json::from_slice(&std::fs::read(preprocessor_config_filename)?)?;
+        let llava_config =
+            hf_llava_config.to_llava_config(&model_name, &generation_config, &preprocessor_config);
+        let tokenizer_filename = api.get("tokenizer.json")?;
+        let tokenizer = Tokenizer::from_file(tokenizer_filename).map_err(E::msg)?;
+        let clip_vision_config = hf_llava_config.to_clip_vision_config();
+        (
+            llava_config,
+            tokenizer,
+            Some(clip_vision_config),
+            preprocessor_config.to_clip_image_processor(),
+        )
+    } else {
+        let config_filename = api.get("config.json")?;
+        let llava_config: LLaVAConfig = serde_json::from_slice(&std::fs::read(config_filename)?)?;
+        println!(
+            "use python to generate tokenizer.json. Will save tokenizer to tokenizer/tokenizer.json"
+        );
+        let cmd = format!("python -c \"from transformers import AutoTokenizer;tokenizer=AutoTokenizer.from_pretrained('{}');tokenizer.save_pretrained('tokenizer')\"", args.model_path);
```

Comment:
Probably better not to rely on python + transformers being available, could you extract it and host it in a specific repo on huggingface, we do that for a bunch of candle models already, e.g. call it `candle-llava`.

Reply:
Since the llava-hf is already existed, create a new repo seems to be meaningless🧐
Here I remove the python download and transform process from code. 
I add them to readme.md. For non-hf model, it is required to download the tokenizer.json manually.